nifi-nar-bundles/nifi-cybersecurity-bundle/nifi-cybersecurity-processors/src/main/java/org/apache/nifi/processors/cybersecurity/FuzzyHashContent.java

```diff
@@ -0,0 +1,192 @@
+import org.apache.nifi.logging.ComponentLog;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.ProcessorInitializationContext;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.io.InputStreamCallback;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.processors.standard.HashContent;
+
+import org.apache.nifi.stream.io.StreamUtils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+
+@EventDriven
+@SideEffectFree
+@SupportsBatching
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@Tags({"hashing", "fuzzy-hashing", "cyber-security"})
+@CapabilityDescription("Calculates a fuzzy/locality-sensitive hash value for the Content of a FlowFile and puts that " +
+        "hash value on the FlowFile as an attribute whose name is determined by the <Hash Attribute Name> property." +
+        "Note: this processor only offers non-criptographic hash algorithms. And it should be not be " +
+        "seen as a replacement to the HashContent processor")
+
+@SeeAlso({HashContent.class})
+@ReadsAttributes({@ReadsAttribute(attribute="", description="")})
+@WritesAttributes({@WritesAttribute(attribute = "<Hash Attribute Name>", description = "This Processor adds an attribute whose value is the result of Hashing the "
+        + "existing FlowFile content. The name of this attribute is specified by the <Hash Attribute Name> property")})
+
+public class FuzzyHashContent extends AbstractProcessor {
+
+    public static final AllowableValue allowableValueSSDEEP = new AllowableValue(
+            "ssdeep",
+            "ssdeep",
+            "Uses ssdeep / SpamSum 'context triggered piecewise hash'.");
+    public static final AllowableValue allowableValueTLSH = new AllowableValue(
+            "tlsh",
+            "tlsh",
+            "Uses TLSH (Trend 'Locality Sensitive Hash'). Note: FlowFile Content must be at least 512 characters long");
+
+    public static final PropertyDescriptor ATTRIBUTE_NAME = new PropertyDescriptor.Builder()
+            .name("ATTRIBUTE_NAME")
+            .displayName("Hash Attribute Name")
+            .description("The name of the FlowFile Attribute into which the Hash Value should be written. " +
+                    "If the value already exists, it will be overwritten")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .defaultValue("fuzzyhash.value")
+            .build();
+
+    public static final PropertyDescriptor HASH_ALGORITHM = new PropertyDescriptor.Builder()
```

Comment:
sounds fair.  Not sure how old HashContent is, may predate EL.  More of a comment as to whether or not this would be useful, but have no bearing for the overall flow and how it would be used.